### PR TITLE
feat(iam): add permissions groups support in iam policy

### PR DESCRIPTION
**Pull request by BrunoMarquesOVH**
ref: MANAGER-12308

| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-12308
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- ~~[ ] Breaking change is mentioned in relevant commits~~

## Description

Add permissions groups support in iam policy

## Related

- [x] CMT-4873

### CODE REVIEW

File: packages/manager/modules/iam/src/components/actionSelect/actionSelect.template.html [hunk 1]
```diff
@@ -28,6 +28,36 @@
...
+            />
+            <label class="oui-checkbox__label-container" for="{{ group.id }}">
+                <span class="oui-checkbox__label">
+                    <span class="oui-checkbox__icon"></span>
+                    <span class="oui-checkbox__text">
+                        <span data-ng-bind="group.name"></span>
+                    </span>
+                </span>
+                <div class="ml-4" data-ng-bind-html="::group.description"></div>
+            </label>
```
Comment (darsene) [inline packages/manager/modules/iam/src/components/actionSelect/actionSelect.template.html:56]: To be consistent you can use one time binding everywhere as you've done for `::group.description`

File: packages/manager/modules/iam/src/components/createPolicy/createPolicy.controller.js [hunk 2]
```diff
@@ -211,6 +211,11 @@ export default class CreatePolicyController {
         (resourceType, index, list) =>
           Boolean(resourceType) && list.indexOf(resourceType) === index,
       );
+      this.model.permissionsGroups = this.policy.permissionsGroups.map(
+        ({ urn }) => {
+          return { urn };
+        },
+      );
```
Comment (darsene) [inline packages/manager/modules/iam/src/components/createPolicy/createPolicy.controller.js:218]:
```suggestion
      this.model.permissionsGroups = this.policy.permissionsGroups.map(
        ({ urn }) => urn,
      );
```
Can be simplified a bit

File: packages/manager/modules/iam/src/components/actionSelect/actionSelect.template.html [hunk 3]
```diff
@@ -28,6 +28,36 @@
         <span data-ng-bind-html="$ctrl.customActionSuccessMessage"></span>
     </oui-message>
 
+    <!-- <MANAGED PERMISSIONS> -->
+    <oui-collapsible
+        data-heading="{{ ::'iam_managed_permissions_heading'  | translate }}"
```
Comment (ghyenne) [inline packages/manager/modules/iam/src/components/actionSelect/actionSelect.template.html:33]:
```suggestion
        data-heading="{{ :: 'iam_managed_permissions_heading' | translate }}"
```

File: packages/manager/modules/iam/src/components/createPolicy/createPolicy.controller.js (hunk 2 above)
Comment (ghyenne) [inline packages/manager/modules/iam/src/components/createPolicy/createPolicy.controller.js:218]:
```suggestion
      this.model.permissionsGroups = this.policy.permissionsGroups.map(
        ({ urn }) => ({ urn }),
      );
```

File: packages/manager/modules/iam/src/policy/edit/edit.routing.js [hunk 4]
```diff
@@ -16,6 +16,14 @@ export default /* @ngInject */ ($stateProvider) => {
...
         return uuid ? IAMService.getDetailedPolicy(uuid) : null;
       },
+
+      /**
+       * Retrieve the list of permissionsGroups
+       * @returns {Object|null}
+       */
+      permissionsGroups: /* @ngInject */ (IAMService) => {
+        return IAMService.getPermissionsGroups();
+      },
```
Comment (darsene) [inline packages/manager/modules/iam/src/policy/edit/edit.routing.js:26]:
```suggestion
      permissionsGroups: /* @ngInject */ (IAMService) => IAMService.getPermissionsGroups(),
```

**Review by ghyenne — APPROVED** [review]

**Review by darsene — CHANGES_REQUESTED** [review]

File: packages/manager/modules/iam/src/components/actionSelect/actionSelect.template.html (hunk 3 above)
Comment (BrunoMarquesOVH) [reply packages/manager/modules/iam/src/components/actionSelect/actionSelect.template.html]: Done

File: packages/manager/modules/iam/src/components/createPolicy/createPolicy.controller.js (hunk 2 above)
Comment (BrunoMarquesOVH) [reply packages/manager/modules/iam/src/components/createPolicy/createPolicy.controller.js]: Done

File: packages/manager/modules/iam/src/components/actionSelect/actionSelect.template.html (hunk 1 above)
Comment (BrunoMarquesOVH) [reply packages/manager/modules/iam/src/components/actionSelect/actionSelect.template.html]: Done

File: packages/manager/modules/iam/src/components/createPolicy/createPolicy.controller.js (hunk 2 above)
Comment (BrunoMarquesOVH) [reply packages/manager/modules/iam/src/components/createPolicy/createPolicy.controller.js]: Done

File: packages/manager/modules/iam/src/policy/create/create.routing.js [hunk 5]
```diff
@@ -7,6 +7,13 @@ export default /* @ngInject */ ($stateProvider) => {
     resolve: {
       breadcrumb: /* @ngInject */ ($translate) =>
         $translate.instant('iam_policy_create'),
+      /**
+       * Retrieve the list of permissionsGroups
+       * @returns {Object|null}
+       */
+      permissionsGroups: /* @ngInject */ (IAMService) => {
+        return IAMService.getPermissionsGroups();
+      },
```
[the remark this answers is not in the document]
Comment (BrunoMarquesOVH) [reply packages/manager/modules/iam/src/policy/create/create.routing.js]: Done

File: packages/manager/modules/iam/src/policy/edit/edit.routing.js (hunk 4 above)
Comment (BrunoMarquesOVH) [reply packages/manager/modules/iam/src/policy/edit/edit.routing.js:26]: Done

Comment (darsene) [reply packages/manager/modules/iam/src/policy/edit/edit.routing.js:26]: seems to be not

**Review by darsene — APPROVED** [review]

**Review by ghyenne — DISMISSED** [review]

**Review by anooparveti — DISMISSED** [review]

**Review by darsene — APPROVED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by dectotam — APPROVED** [review]

**Review by darsene — APPROVED** [review]

**Pull request merged**